src/object.d

```diff
@@ -1890,6 +1891,16 @@ void* aaLiteral(Key, Value)(Key[] keys, Value[] values) @trusted pure
 
 alias AssociativeArray(Key, Value) = Value[Key];
 
+void clear(T : Value[Key], Value, Key)(T aa)
+{
+    _aaClear(*cast(void **)&aa);
+}
+
+void clear(T : Value[Key], Value, Key)(T* aa)
```

Comment:
I don't know, I was following precedent, the other methods (e.g. rehash, dup) also do this.